azkaban-common/src/main/java/azkaban/database/DataSourceUtils.java

```diff
@@ -169,6 +169,7 @@ public String getDBType() {
 
       public MonitorThread(MySQLBasicDataSource mysqlSource) {
         this.setName("MySQL-DB-Monitor-Thread");
+        setDaemon(true);
```

Comment:
Don't think so. Note that this change will not fix the shutdown hang problem.

However, the monitor thread isn't something that we need to hold up the executor for. That's the reason I marked it as a daemon thread.